sdk/python/kfp/compiler/compiler.py

```diff
@@ -820,23 +817,23 @@ def _create_workflow(self,
     self._sanitize_and_inject_artifact(dsl_pipeline, pipeline_conf)
 
     # Fill in the default values.
-    args_list_with_defaults = []
```

Comment:
What is the motivation of changing this from List to OrderedDict? Seems to me we still only iterate through it as if it's a sequential structure. I don't see any usage like key-lookup or similar. Did I miss anything?

Reply:
`args_list_with_defaults` is passed to `_create_pipeline_workflow` which now expects a dictionary, not a list of `PipelineParam`s. Line 629 iterates through the keys and values of the dictionary.